### PR TITLE
Adds API to customize the rendered navigation dots.

### PR DESCRIPTION
**Pull request by nkemcels**
This is a very common use case, one which I badly needed. To customize the navigation dots for the carousel. I forked the repository, did the necessary changes and decided to help others that will need it through this pull request. 
If there are any other things to modify before merging the pull request, I'll gladly take on that.

### CODE REVIEW

**Review by maxmarinich — COMMENTED** [review]
Hi, @lehos! Improvements needed.

File: src/lib/react-alice-carousel.tsx [hunk 1]
```diff
@@ -575,7 +575,13 @@ export default class AliceCarousel extends React.PureComponent<Props, State> {
 	};
 
 	_renderDotsNavigation() {
-		return <Views.DotsNavigation state={this.state} onClick={this._handleDotClick} />;
+		const {renderDotsNavigation} = this.props;
+		if(typeof renderDotsNavigation == "function"){
+			const {activeIndex, itemsCount} = this.state;
+			return renderDotsNavigation(itemsCount, activeIndex);
```
[the remark this answers is not in the document]
Comment (nkemcels) [reply src/lib/react-alice-carousel.tsx:581]: Hi @maxmarinich, Actually, if a user wants to render a custom navigation dot, then they will have to handle the click events by themselves. Besides, they don't have to render dots which was the point of the pull request. They could render something custom like display the current slide number etc.

[the remark this answers is not in the document]
Comment (maxmarinich) [reply src/lib/react-alice-carousel.tsx:581]: We need to provide a complete API with the expected behavior. Please check #148 if interested. The complete sample is [here](https://maxmarinich.github.io/react-alice-carousel/static/#custom-components). Аvailable from [v2.1.0](https://www.npmjs.com/package/react-alice-carousel).

**Comment by maxmarinich** [comment]
Implemented in #148.

**Pull request closed without merging**